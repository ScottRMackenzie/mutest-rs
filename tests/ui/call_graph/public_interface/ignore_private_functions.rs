//@ print-call-graph
//@ print-targets
//@ stdout
//@ stderr: empty
//@ mutest-flags: --crate-kind=mutant-for-external-tests

#[allow(unused)]
fn private_nested_fn() {}

fn private_reachable_fn() {}

#[allow(unused)]
fn private_fn() {
    private_nested_fn();
    private_reachable_fn();
}

pub fn public_and_nested_fn() {
    private_reachable_fn();
}

pub fn public_fn() {
    public_and_nested_fn();
    private_reachable_fn();
}
