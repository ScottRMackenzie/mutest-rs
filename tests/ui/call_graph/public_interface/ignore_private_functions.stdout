
@@@ call graph @@@

entry points:

public public_and_nested_fn
  -> private_reachable_fn at tests/ui/call_graph/public_interface/ignore_private_functions.rs:10:1: 10:26 (#0)
public public_fn
  -> private_reachable_fn at tests/ui/call_graph/public_interface/ignore_private_functions.rs:10:1: 10:26 (#0)
  -> public_and_nested_fn at tests/ui/call_graph/public_interface/ignore_private_functions.rs:18:1: 18:30 (#0)

nested calls at distance 1:

public_and_nested_fn at tests/ui/call_graph/public_interface/ignore_private_functions.rs:18:1: 18:30 (#0)
  -> private_reachable_fn at tests/ui/call_graph/public_interface/ignore_private_functions.rs:10:1: 10:26 (#0)

@@@ targets @@@

public interface -(0)-> private_reachable_fn at tests/ui/call_graph/public_interface/ignore_private_functions.rs:10:1: 10:26 (#0)
  (0) public_and_nested_fn
  (0) public_fn

public interface public_and_nested_fn at tests/ui/call_graph/public_interface/ignore_private_functions.rs:18:1: 18:30 (#0)
  (0) public_fn

public interface public_fn at tests/ui/call_graph/public_interface/ignore_private_functions.rs:22:1: 22:19 (#0)

targets: 3 total; 3 safe; 0 unsafe (0 tainted)
